src/ui/components/VPNConnectionStability.qml

```diff
@@ -72,6 +73,21 @@ RowLayout {
         }
     ]
 
+    transitions: [
```

Comment:
It could be that this qml file is not rendered (for instance, if we have the settings opened). What about if we trigger this event from C++?
Instead of this block of code, you can add here: https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/src/connectionhealth.cpp#L68

```
if (stability == Unstable) {
  emit MozillaVPN::instance()->triggerGleanSample(GleanSample::connectionHealthUnstable);
} else if (stability == NoSignal) {
  emit MozillaVPN::instance()->triggerGleanSample(GleanSample::connectionHealthNoSignal);
}
```

At the beginning of the same file, here: https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/src/connectionhealth.cpp#L6 (alphabetic order) add:

```
#include "gleansample.h"
```